pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -427,14 +428,31 @@ func (r *ReconcileCloneSet) getActiveRevisions(cs *appsv1alpha1.CloneSet, revisi
 		return nil, nil, collisionCount, err
 	}
 
+	// Here, we do not consider the case where the volumeClaimTemplates Hash was not added before the upgrade feature.
+	// 1. It is not possible to actually extract the previous volumeClaimTemplates Hash.
+	// 2. the hash must exist in all revisions that have been updated after the feature is enabled.
+	EqualVCTHashEqual := func(revision *apps.ControllerRevision, needle *apps.ControllerRevision) bool {
+		needleVolumeClaimHash, _ := volumeclaimtemplate.GetVCTemplatesHash(needle)
+		tmpHash, exist := volumeclaimtemplate.GetVCTemplatesHash(revision)
+		// vcTemplate hash may not exist in old revision, use hash of new revision instead
+		if !exist || tmpHash != needleVolumeClaimHash {
+			return false
+		}
+		return true
+	}
+
 	// find any equivalent revisions
 	equalRevisions := history.FindEqualRevisions(revisions, updateRevision)
 	equalCount := len(equalRevisions)
-
 	if equalCount > 0 && history.EqualRevision(revisions[revisionCount-1], equalRevisions[equalCount-1]) {
```

Comment:
Let's add a comment that states that only PVC changes are not newly generated revisions.